x-pack/filebeat/input/azureblobstorage/job.go

```diff
@@ -222,8 +230,11 @@ func (j *job) createEvent(message string, offset int64) beat.Event {
 					},
 				},
 			},
-			"cloud": mapstr.M{
-				"provider": "azure",
+			// Structs are used here in order to save map allocations
```

Comment:
Why limit this just to this field? The fields above look like they could benefit from the same optimisation. Though I'd suggest that this optimisation be done in a separate, non-bugfix change.